docs/DeveloperGuide.md

```diff
@@ -1,33 +1,244 @@
+&nbsp;&nbsp;&nbsp;&nbsp;[Scope](#scope) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;[Design Goals](#design-goals) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;[Maintainability](#maintainability) <br>
+[Definitions](#definitions) <br>
+[Setting up the project in your computer](#setting-up-the-project-in-your-computer) <br>
+[System Overview](#system-overview) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;[Technological view](#technological-view) <br>
+[Architecture](#architecture) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;[Project Overview](#project-overview) <br>
+[Writings class family](#writings-class-family) <br>
+[Filter words class family](#filter-words-class-family) <br>
+[Bunny class family](#bunny-class-family) <br>
+[Names class family](#names-class-family) <br>
+[Design and Implementation](#design-and-implementation) <br>
+[Product scope](#product-scope) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;[Target user profile](#target-user-profile) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;[Value proposition](#value-proposition) <br>
+[User Stories](#user-stories) <br>
+[Non-Functional Requirements](#non-functional-requirements) <br>
+[Glossary](#glossary) <br>
+[Instruction for manual testing](#instructions-for-manual-testing) <br>
+
+## Introduction
+
+### Purpose
+This document specified architecture and software design decisions for the creative writing assistant, Fluffle. 
+
+### Scope
+This describes the software architecture and software design decisions for the implementation of Fluffle. The intended 
+audience of this document is the developers, designers, and software testers of Fluffle.
+
+### Design Goals
+Creating an app which improves users’ ability of writing and creativity. Developers should work closely with CS2113T’s 
+module instructors, who represent technical advisors and CS2101’s module instructors, who represent non-technical 
+advisors. By adhering to these tutors’ specific requirements, which imitate industrial professional standards, 
+beginner software engineers will be able to levitate their expertises and mindset in the process of developing 
+and presenting a new product.
+
+### Maintainability
+With the aim of increasing maintainability of Fluffle, separated packages and classes were implemented in strict 
+compliance with Object-oriented Programming. Since it is important to be able to easily change functionality of one 
+class without disturbing other dependent classes, each class only serves one purpose to reduce dependency on other 
+classes. With low coupling and high cohesion, subsequent developers can make minimal effort in maintaining Fluffle.
+
+## Definitions
+
+|Terminology|Explanation|
+|--------|----------|
+|Plot bunny|A story idea that refuses to go away until it is written down.|
+|Fluffle|A fluffle is a group of bunnies.| 
+|CLI|Command-line Interface. </br></br>A command-line interface (CLI) processes command to a computer program in the form of lines of text. The program in the computer will read the command and start proceeding.|
```

Comment:
Do note there seems to be a issue with the formatting for the '/br' for the github pages generated DG